client/homebrew/brewRenderer/safeHTML.js

```diff
@@ -0,0 +1,47 @@
+let doc = null;
+let div = null;
+
+function safeHTML(htmlString) {
+	// If the Document interface doesn't exist, exit
+	if(!document) return null;
+	// If the test document and div don't exist, create them
+	if(!doc) doc = document.implementation.createHTMLDocument('');
+	if(!div) div = doc.createElement('div');
+
+	// Set the test div contents to the evaluation string
+	div.innerHTML = htmlString;
+	// Grab all nodes from the test div
+	const elements = div.querySelectorAll('*');
+
+	// Blacklisted tags
+	const blacklistTags = ['script', 'noscript', 'noembed'];
+	// Tests to remove attributes
+	const blacklistAttrs = [
+		(test)=>{return test.localName.indexOf('on') == 0;},
+		(test)=>{return test.value.replace(/[\u0000-\u0020\u00A0\u1680\u180E\u2000-\u2029\u205f\u3000]/g, '').toLowerCase().trim().indexOf('javascript:') == 0;}
+	];
+
+
+	elements.forEach((element)=>{
+		// Check each element for blacklisted type
+		if(blacklistTags.includes(element?.localName?.toLowerCase())) {
+			element.parentNode.removeChild(element);
+			return;
+		}
+		// Check remaining elements for blacklisted attributes
+		if(element.hasAttributes()){
+			for (const attribute of element.attributes){
+				for (const test of blacklistAttrs) {
+					if(test(attribute)){
+						element.removeAttribute(attribute.localName);
+						break;
+					};
+				};
+			};
+		};
```

Comment:
I'm not super bothered by it, but this stands out as a lot of nested if's and for's.

Can you use `[...element.attributes].forEach()` rather than the first `for(const attribute....`?

And then for the second `for` loop, use `if(blacklistAttrs.some(test=>test(attribute))){ remove.... }`

Reply:
> `[...element.attributes].forEach()`

I don't believe you can `forEach` works on a NodeList, but this would convert it to an array first, so it might work.

> `if(blacklistAttrs.some(test=>test(attribute))){ remove.... }`

That will probably work; I keep forgetting `some()` exists and finding harder ways to achieve the same result.

I'll experiment when I have some time; might be a few days though.